sdk/trace/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -57,7 +56,7 @@
   // The kind of the span.
   private final SpanKind kind;
   // The clock used to get the time.
-  private final Clock clock;
+  private final MonotonicClock clock;
```

Comment:
Given `RecordEventsReadableSpan` wouldn't work correctly with any other `Clock` this seems like an unexpectedly big improvement